### ISSUE TITLE
MeshAgent not connecting over VPN

### ISSUE THREAD

**Issue by zaMuffenMan** [Ylianst/MeshCentral#3394]
Hi team, 

I've been running a MeshCentral server (on a RaspberryPi) on a local network (Network A) with multiple MeshAgents and everything is connecting and working fine on this network. I have another MeshAgent installed on a remote RaspberryPi (Network B). Network B is connected to Network A over a Wireguard VPN.

I have my MeshCentral server running in Hybrid mode, with the MeshAgent on Network B connecting using

`MeshServer=wss://{MeshCentral IP Address}:443/agent.ashx`

And this was working fine for a few weeks until it suddenly stopped working a few days ago.

I've rebooted both the MeshCentral, VPN Server and remote computer. I've uninstalled and reinstalled the MeshAgent - still the MeshAgent will not connect.

Doing a bit more debugging, when connecting the MeshAgent manually on the remote computer I get this output:

`Connecting to: wss://{MeshCentral IP Address}:443/agent.ashx`
`2022-01-01 08:22:39 PM: Control Channel Idle Timeout = 120 seconds`
`Connected.`
`Timeout waiting for Server, launching cached meshcore...`
`Mesh Server Connection Error [12]`
`Connecting to: wss://{MeshCentral IP Address}:443/agent.ashx`
`2022-01-01 08:24:45 PM: Control Channel Idle Timeout = 120 seconds`
`Connected.`
`Mesh Server Connection Error [16]`
`Connecting to: wss://{MeshCentral IP Address}:443/agent.ashx`
`2022-01-01 08:26:52 PM: Control Channel Idle Timeout = 120 seconds`
`Connected.`

While on the sever trace I get:

`20:24:45 - AGENT: New agent at {VPN Server Address}:44563`
`20:24:45 - WEBREQUEST: ({VPN Server Address}) /agent.ashx/.websocket`
`20:22:39 - AGENT: New agent at {VPN Server Address}:53863`
`20:22:39 - WEBREQUEST: ({VPN Server Address}) /agent.ashx/.websocket`

Any advise what I could do to get it back up and running?

Thanks!

**Comment by Ylianst** [comment]
Weird. I have not seen this before. The agent connects but the server does not do anything. I am going to take a crazy guess and ask you to restore a older version of the database if you can. In "meshcentral-backups" you should have older versions, copy these offsite and restore the "*.db" files in "meshcentral-data" to older versions.

I am going to take a wild guess that your using NeDB and the DB got corrupt.

**Comment by zaMuffenMan** [comment]
Thanks for the quick response. I am using NeDB due to the limited number of devices.

I tried the above, but unfortunately it still doesn't seem to work. To recap what I did:

I ran `sudo systemctl stop meshcentral` on the server, copied the backup "*.db" files in "meshcentral-backup" from a backup prior to the meshagent failing to "meshcentral-data".

Ran `sudo systemctl start meshcentral` turned on tracing and saw the same output as above with the agent still showing as offline.

I removed the agent from MeshCentral and it doesn't register it again. 

Checked the agent output on the remote device and the same output was appearing there as well...
